src/main/java/org/broadinstitute/hellbender/tools/walkers/featuremapping/AddFlowSNVQuality.java

```diff
@@ -0,0 +1,601 @@
+
+        // create a key slice of the area around the flow/hmer.
+        final int minIndex = Math.max(flow - (flowOrderLength - 1), 0);
+        final int maxIndex = Math.min(flow + (flowOrderLength - 1), key.length - 1);
+        final int[] slice = Arrays.copyOfRange(key, minIndex, maxIndex + 1);
+        final int hmerLength = key[flow];
+        final double[] p12 = new double[2];
+
+        // walk the flows towards the side until (and including) the first non-zero key
+        // on hmers of length 1 we walk both sides
+        final class SliceInfo {
+            int[] slice;
+            byte altByte;
+            int sideFlow;
+        }
+        final List<SliceInfo> slices = new LinkedList<>();
+        final int[]           incrs = (hmerLength != 1)
+                ? new int[] { sideIncr }
+                : new int[] { sideIncr, -sideIncr};
+        for (int incr : incrs) {
+            for (int sideFlow = flow + incr; sideFlow >= 0 && sideFlow < key.length; sideFlow += incr) {
+
+                // side flow can no overflow the slice
+                if ( sideFlow < minIndex || sideFlow > maxIndex ) {
+                    break;
+                }
+
+                // create a alternative key slice by incrementing sideFlow and decrementing flow
+                final int[] altSlice = Arrays.copyOf(slice, slice.length);
+                altSlice[sideFlow - minIndex] += 1;
+                altSlice[flow - minIndex] -= 1;
+                if ( sliceIsValid(altSlice, flowOrderLength) ) {
+                    SliceInfo si = new SliceInfo();
+                    si.slice = altSlice;
+                    si.altByte = flowOrder[sideFlow % flowOrderLength];
+                    si.sideFlow = sideFlow;
+                    slices.add(si);
+                }
+
+                // is the sideFlow (the first encountered) non-zero? if so, break
+                if (key[sideFlow] != 0) {
+                    break;
+                }
+            }
+        }
+
+        // at this point, we have a list of valid slices. figure out the error probability for each of them
+        // and compute the base quality
+        final double keyP = sliceProb(slice, minIndex, key, errorProbBands, flow, flow, null);
+        double sumP = keyP;
+        for ( final SliceInfo si : slices ) {
+            final double sliceP = sliceProb(si.slice, minIndex, key, errorProbBands, flow, si.sideFlow, p12);
+            if ( allBaseProb != null ) {
+                allBaseProb.put(si.altByte, getSnvq(sliceP, p12[0], p12[1]));
+            }
+            sumP += sliceP;
+        }
+        final double ep = 1 - (keyP / sumP);
+
+        return ep;
+    }
+
+    private double getSnvq(final double sliceP, final double p1, final double p2) {
+        if ( aqArgs.snvMode == AddFlowSNVQualityArgumentCollection.SnvqModeEnum.Legacy ) {
```

Comment:
What is this?

Reply:
this is resolved by removing of all the multi-strategy code.